File: b884(jack_in_computer_classroom)/C++/jack_in_computer_classroom.cpp
#include <iostream>

int main(int argc, char const *argv[]) {
  int num_of_cases, x, y, yee;
  while (std::cin >> num_of_cases) {
    while (num_of_cases--) {
      std::cin >> x >> y;
      yee = 100 - x - y;
      if (0 < yee && yee <= 30) std::cout << "sad!\n";
      else if (30 < yee && yee <= 60) std::cout << "hmm~~\n";
      else if (60 < yee && yee < 100) std::cout << "Happyyummy\n";
      else std::cout << "evil!!\n";
    }
  }
  return 0;
}

File: b884(jack_in_computer_classroom)/Python/jack_in_computer_classroom.py
from sys import stdin, stdout

while True:
    num_of_cases = stdin.readline().strip('\n')
    if not num_of_cases: break
    for c in range(int(num_of_cases)):
        x, y = map(int, stdin.readline().split())
        yee = 100 - x - y
        if 0 < yee <= 30:
            stdout.write('sad!\n')
        elif 30 < yee <= 60:
            stdout.write('hmm~~\n')
        elif 60 < yee < 100:
            stdout.write('Happyyummy\n')
        else:
            stdout.write('evil!!\n')
